dashboard/modules/runtime_env/runtime_env_agent.py

```diff
@@ -59,6 +59,104 @@ class CreatedEnvResult:
     result: str
 
 
+class UriType(Enum):
+    WORKING_DIR = 1
+    PY_MODULES = 2
+    PIP = 3
+    CONDA = 4
+
+
+class ReferenceTable:
+    """
+    The URI reference table which is used for GC.
+    When the reference count is decreased to zero,
+    the URI should be removed from this table and
+    added to cache if needed.
+    """
+
+    def __init__(
+        self,
+        uris_parser: Callable[[RuntimeEnv], None],
+        unused_uris_callback: Callable[[List[Tuple[UriType, str]]], None],
+        unused_runtime_env_callback: Callable[[str], None],
+    ):
+        self._logger = default_logger
+        self._uri_reference: Dict[str, int] = dict()
+        self._runtime_env_reference: Dict[str, int] = dict()
```

Comment:
Do we need it? Isn't uri reference enough?

Reply:
As the comment https://github.com/ray-project/ray/pull/22828#issuecomment-1067532865 says, I think an additional `_runtime_env_reference` could make the logic more clear.

Reply:
@architkulkarni why do we need uri_to_env at all? There's no meaning to follow env anymore right? We just need to track of uris here.